packages/api/cms-api/src/dam/files/files.service.ts

```diff
@@ -385,6 +401,180 @@ export class FilesService {
+            for (const file of files) {
+                if (file.scope === undefined) {
+                    continue;
+                }
+
+                const isDuplicateScope = Boolean(fileScopes.find((scope) => this.contentScopeService.scopesAreEqual(scope, file.scope)));
+                if (!isDuplicateScope) {
+                    fileScopes.push(file.scope);
+                }
+            }
+            return fileScopes;
+        };
+
+        const canAccessFileScopes = (fileScopes: DamScopeInterface[], user: CurrentUserInterface) => {
+            for (const scope of fileScopes) {
+                const canAccessFileScope = this.contentScopeService.canAccessScope(scope, user);
+                if (!canAccessFileScope) {
+                    return false;
+                }
+            }
+            return true;
+        };
+
+        const getOrCreateTargetFolder = async ({
+            targetFolderId,
+            targetScope,
+            fileScopes,
+        }: {
+            targetFolderId?: string;
+            targetScope: DamScopeInterface;
+            fileScopes: DamScopeInterface[];
+        }) => {
+            let createdTargetFolderAutomatically = false;
+            let targetFolder: FolderInterface;
+            if (targetFolderId) {
+                const folder = await this.foldersService.findOneById(targetFolderId);
+                if (folder === null) {
+                    throw new Error("Specified target folder doesn't exist.");
+                }
+
+                targetFolder = folder;
+            } else {
+                const scopeString = fileScopes.length === 0 ? "unknown" : fileScopes.map((scope) => Object.values(scope).join("-")).join(", ");
+                const date = new Date();
+
+                targetFolder = await this.foldersService.create(
+                    {
+                        name: `Copy from ${scopeString} ${format(date, "dd.MM.yyyy")}, ${format(date, "HH:mm:ss")}`,
+                        isInboxFromOtherScope: true,
+                    },
+                    targetScope,
+                );
+                createdTargetFolderAutomatically = true;
+            }
+
+            return { targetFolder, createdTargetFolderAutomatically };
+        };
+
+        const findCopyWithSameCroppingInTargetScope = async (file: FileInterface) => {
+            const copiesInTargetScope = await this.findCopiesOfFileInScope(file.id, targetScope);
+
+            if (copiesInTargetScope.length === 0) {
+                return undefined;
+            }
+
+            for (const copy of copiesInTargetScope) {
+                if (isEqual(file.image?.cropArea, copy.image?.cropArea)) {
+                    return copy;
+                }
+            }
+
+            return undefined;
+        };
+
+        if (!this.contentScopeService.canAccessScope(targetScope, user)) {
+            throw new ForbiddenException("User can't access the target scope");
```

Comment:
This is an HTTP exception, which has no meaning in GraphQL operations. You might simply throw a normal error here. But we still have to come up with a best practice on how to handle errors in GraphQL.